src/module.rs

```diff
@@ -321,59 +321,56 @@ impl<'ctx> Module<'ctx> {
         Some(BasicTypeEnum::new(type_))
     }
 
-    /// Sets a `Target` to this `Module`.
+    /// Assigns a `TargetTriple` to this `Module`.
     ///
     /// # Example
     ///
     /// ```rust,no_run
     /// use inkwell::context::Context;
-    /// use inkwell::targets::Target;
-    ///
-    /// Target::initialize_x86(&Default::default());
+    /// use inkwell::targets::TargetTriple;
     ///
     /// let context = Context::create();
     /// let module = context.create_module("mod");
-    /// let target = Target::from_name("x86-64").unwrap();
+    /// let triple = TargetTriple::create("x86_64-pc-linux-gnu");
     ///
-    /// assert!(module.get_target().is_none());
+    /// assert_eq!(module.get_target(), TargetTriple::create(""));
     ///
-    /// module.set_target(&target);
+    /// module.set_target(&triple);
     ///
-    /// assert_eq!(module.get_target().unwrap(), target);
+    /// assert_eq!(module.get_target(), triple);
     /// ```
-    pub fn set_target(&self, target: &Target) {
+    pub fn set_target(&self, target_triple: &TargetTriple) {
         unsafe {
-            LLVMSetTarget(self.module.get(), target.get_name().as_ptr())
+            LLVMSetTarget(self.module.get(), target_triple.as_ptr())
         }
     }
 
-    /// Gets the `Target` assigned to this `Module`, if any.
+    /// Gets the `TargetTriple` assigned to this `Module`. If none has been
+    /// assigned, the triple will default to "".
     ///
     /// # Example
     ///
     /// ```rust,no_run
     /// use inkwell::context::Context;
-    /// use inkwell::targets::Target;
-    ///
-    /// Target::initialize_x86(&Default::default());
+    /// use inkwell::targets::TargetTriple;
     ///
     /// let context = Context::create();
     /// let module = context.create_module("mod");
-    /// let target = Target::from_name("x86-64").unwrap();
+    /// let triple = TargetTriple::create("x86_64-pc-linux-gnu");
     ///
-    /// assert!(module.get_target().is_none());
+    /// assert_eq!(module.get_target(), TargetTriple::create(""));
     ///
-    /// module.set_target(&target);
+    /// module.set_target(&triple);
     ///
-    /// assert_eq!(module.get_target().unwrap(), target);
+    /// assert_eq!(module.get_target(), triple);
     /// ```
-    pub fn get_target(&self) -> Option<Target> {
+    pub fn get_target(&self) -> TargetTriple {
         // REVIEW: This isn't an owned LLVMString, is it? If so, need to deallocate.
         let target_str = unsafe {
             LLVMGetTarget(self.module.get())
         };
 
-        Target::from_name_raw(target_str)
+        TargetTriple::new_borrowed(target_str)
```

Comment:
A big problem is that we're leaking a raw pointer into the wild. Really this should be `-> &TargetTriple` for borrowed triples

Reply:
I think a similar problem may exist with DataLayout since the origin of the problem is really with LLVMStringOrRaw not having a proper lifetime on its borrow variant. Maybe the solution is to define an LLVMStr that implements ToOwned and stick it inside an actual Cow?

Reply:
I fixed by making a copy. I can't imagine that this is a hot path anyway and doing something more efficient really increased code complexity. Similarly, there are more efficiently ways to implement `TargetTriple;:create` (maybe you could hold an enum of `LLVMString` and `CString`) but I'm not really sure they're worth it. Also, as mentioned above, there is definitely a leakage issue with `DataLayout::get_borrowed_data_layout`. I guess I will make a separate issue for this...

Reply:
Sounds good. We can always optimize the copy out in the future. Please just add a TODO comment so that we don't forget.

Reply:
And maybe an issue too